File: CS/DXGridRowStateHelper/Models/Customer.cs
using System.ComponentModel;

namespace DXGridRowStateHelper.Models
{
    public class Customer : INotifyPropertyChanged
    {
        #region CustomerId Property

        private int m_CustomerId = 0;
        public int CustomerId
        {
            get
            {
                return this.m_CustomerId;
            }
            set
            {
                this.m_CustomerId = value;

                this.OnPropertyChanged("CustomerId");
            }
        }

        #endregion

        #region RandomId Property

        private int m_RandomId = 0;
        public int RandomId
        {
            get
            {
                return this.m_RandomId;
            }
            set
            {
                this.m_RandomId = value;

                this.OnPropertyChanged("RandomId");
            }
        }

        #endregion

        #region RandomId2 Property

        private int m_RandomId2 = 0;
        public int RandomId2
        {
            get
            {
                return this.m_RandomId2;
            }
            set
            {
                this.m_RandomId2 = value;

                this.OnPropertyChanged("RandomId2");
            }
        }

        #endregion

        #region FirstName Property

        private string m_FirstName = string.Empty;
        public string FirstName
        {
            get
            {
                return this.m_FirstName;
            }
            set
            {
                this.m_FirstName = value;

                this.OnPropertyChanged("FirstName");
            }
        }

        #endregion

        #region LastName Property

        private string m_LastName = string.Empty;
        public string LastName
        {
            get
            {
                return this.m_LastName;
            }
            set
            {
                this.m_LastName = value;

                this.OnPropertyChanged("LastName");
            }
        }

        #endregion

        public event PropertyChangedEventHandler PropertyChanged;

        private void OnPropertyChanged(string propertyName)
        {
            var handler = this.PropertyChanged;
            if (handler != null)
            {
                handler(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}

File: VB/DXGridRowStateHelper/Models/Customer.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel

Namespace DXGridRowStateHelper.Models
	Public Class Customer
		Implements INotifyPropertyChanged
		#Region "CustomerId Property"

		Private m_CustomerId As Integer = 0
		Public Property CustomerId() As Integer
			Get
				Return Me.m_CustomerId
			End Get
			Set(ByVal value As Integer)
				Me.m_CustomerId = value

				Me.OnPropertyChanged("CustomerId")
			End Set
		End Property

		#End Region

		#Region "RandomId Property"

		Private m_RandomId As Integer = 0
		Public Property RandomId() As Integer
			Get
				Return Me.m_RandomId
			End Get
			Set(ByVal value As Integer)
				Me.m_RandomId = value

				Me.OnPropertyChanged("RandomId")
			End Set
		End Property

		#End Region

		#Region "RandomId2 Property"

		Private m_RandomId2 As Integer = 0
		Public Property RandomId2() As Integer
			Get
				Return Me.m_RandomId2
			End Get
			Set(ByVal value As Integer)
				Me.m_RandomId2 = value

				Me.OnPropertyChanged("RandomId2")
			End Set
		End Property

		#End Region

		#Region "FirstName Property"

		Private m_FirstName As String = String.Empty
		Public Property FirstName() As String
			Get
				Return Me.m_FirstName
			End Get
			Set(ByVal value As String)
				Me.m_FirstName = value

				Me.OnPropertyChanged("FirstName")
			End Set
		End Property

		#End Region

		#Region "LastName Property"

		Private m_LastName As String = String.Empty
		Public Property LastName() As String
			Get
				Return Me.m_LastName
			End Get
			Set(ByVal value As String)
				Me.m_LastName = value

				Me.OnPropertyChanged("LastName")
			End Set
		End Property

		#End Region

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Private Sub OnPropertyChanged(ByVal propertyName As String)
			Dim handler = Me.PropertyChangedEvent
			If handler IsNot Nothing Then
				handler(Me, New PropertyChangedEventArgs(propertyName))
			End If
		End Sub
	End Class
End Namespace